20221104 ad review (#54)
NLnetLabs/draft-toorop-dnsop-dns-catalog-zones: draft-ietf-dnsop-dns-catalog-zones.md

```

In this scenario, consumer(s) shall not sign the member zone "example.com." with
DNSSEC.
For "example.net.", the consumers, at two different operators, shall configure

> the member zone to be signed with a NSEC3 chain.  The group value that indicates

Reviewer: ```suggestion
the member zone to be signed with an NSEC3 chain.  The group value that indicates
```

(also, would like to echo Libor's sentiment that non-wrapped sentences are way more fun in diffs)

> that depends on what's agreed with each operator ("operator-x-sign-with-nsec3"

Reviewer: ```suggestion
that depends on what has been agreed with each operator ("operator-x-sign-with-nsec3"
```

# Implementation and operational Notes {#implementationnotes}

> Although any valid domain name can be used for the catalog name $CATZ, it is

Reviewer 2: ```suggestion
Although any valid domain name can be used for the catalog name $CATZ, a catalog producer MUST NOT use names that are not under the control of the catalog producer. It is
```

RECOMMENDED to use either a domain name controlled by the catalog producer, or

> to use a name under a suitable name such as "invalid." [@!RFC6761].

Reviewer: I preferred the less specific version (Special-Use)

Author: How do you recommend to address the following comment from the AD review:
> Also, what is a "suitable Special-Use Domain Name"? 10.in-addr.arpa.? localhost.? example.com.? Why doesn't this just say that the catalog name MUST be under a domain owned by the catalog producer?

Reviewer 2: Special-Use Domain Names are listed here: https://www.iana.org/assignments/special-use-domain-names/special-use-domain-names.xhtml.
From the names listed in RFC6761, "invalid." is indeed the most suitable. But there may be future Special-Use domain names (".alt" or ".internal") that may be suitable too. If needed we can register something specially for catalog zones, something like "catalog-zones.arpa." maybe.

Author: Do we have an answer to Warren's question "Why doesn't this just say that the catalog name MUST be under a domain owned by the catalog producer?"

Reviewer 2: Yes, the catalog zone is not (necessarily) part of the global name space. It is part of the control plane and not the data plane. An unpublished DNS domain (other than to serve the catalog zone) makes therefore sense (at least to me), also to not accidentally leak the zones in the catalog (which may be private).

Reviewer 2: > I preferred the less specific version (Special-Use)

I think "such as" is also not too specific. So I'll go ahead and merge.

> to use a name under a suitable name such as "invalid." [@!RFC6761].

Author: > Why replace "owned" by "controlled"?

Because I don't think you can "own" a name under `invalid.`. What matters is that the producer controls it in the DNS context at hand. (Similarly, it's sufficient for the ACME protocol that a certificate requestor controls the name.)

> Also, reintroducing Special-Use domain name (but keeping the suggestion for "invalid.") to allow for future domain names under ".alt"

My impression was that Warren was not happy about mentioning special-use names (`what is a "suitable Special-Use Domain Name"?` -- note that his question was not simply what "suitable" means). I think this is because the ".alt" discussion has sparked uncertainty about the future of RFC 6761.

I therefore thought that any name deemed suitable in the local context can serve for this, notwithstanding use of future domain names under ".alt", and only vaguely referring to the RFC via the reference.

These were my attempts to "water down" the specificity of normative statements in a pragmatic way so that the purpose is served, while forestalling potential discussion about the role of RFC 6761 names. -- However, I have no stakes at all for the actual wording, and am happy to concede to whatever the group deems best.

> to use a name under a suitable name such as "invalid." [@!RFC6761].

Reviewer 2: > > Why replace "owned" by "controlled"?
> 
> Because I don't think you can "own" a name under `invalid.`. 

No, but is says "or". So either "owned" or "something under invalid."

> What matters is that the producer controls it in the DNS context at hand. (Similarly, it's sufficient for the ACME protocol that a certificate requestor controls the name.)

Ack. But part of the intention of the paragraph was to prevent hijacking domain name space (as we have done in our testbeds b.t.w. with the "catalog." tld.). Therefore the emphasis on either "owned" or "specially reserved"

> > Also, reintroducing Special-Use domain name (but keeping the suggestion for "invalid.") to allow for future domain names under ".alt"
> 
> My impression was that Warren was not happy about mentioning special-use names (`what is a "suitable Special-Use Domain Name"?` -- note that his question was not simply what "suitable" means). I think this is because the ".alt" discussion has sparked uncertainty about the future of RFC 6761.

> I therefore thought that any name deemed suitable in the local context can serve for this, notwithstanding use of future domain names under ".alt", and only vaguely referring to the RFC via the reference.
> 
> These were my attempts to "water down" the specificity of normative statements in a pragmatic way so that the purpose is served, while forestalling potential discussion about the role of RFC 6761 names. -- However, I have no stakes at all for the actual wording, and am happy to concede to whatever the group deems best.
